### PR TITLE
Add build-forc-test-project to ci

### PR DESCRIPTION
**Pull request by ra0x3**
### Description
- This PR adds a `build-forc-test-project` CI job that simply builds, formats, and tests a new forc project to ensure basic commands aren't broken

### Testing steps
- [ ] Added `build-forc-test-project` CI job should pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -127,7 +127,7 @@ jobs:
       - name: Build test project
         run: forc build --path test-proj
       - name: Run test project's test suite
-        run: furc test --path test-proj
```
Comment (mohammadfawaz) [inline .github/workflows/ci.yml]: Ok, `furc test` sounds awesome.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -102,6 +102,33 @@ jobs:
             rm tmp.txt && exit 0
           fi
 
+  build-forc-test-project:
```
Comment (mitchmindtree) [inline .github/workflows/ci.yml:105]: Could you add this new job to the list of `needs` for each of the publish jobs?

**Review by mitchmindtree — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -437,6 +464,7 @@ jobs:
   build-publish-master-image:
     needs:
       [
+        build-forc-test-project,
```
Comment (mitchmindtree) [inline .github/workflows/ci.yml:467]: Could you also add this to the `needs` arrays for the `notify-slack-on-failure` and `publish` jobs?

**Review by mitchmindtree — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (mitchmindtree) [reply .github/workflows/ci.yml:468]: @ra0x3 we're still missing this under the `publish` job.

**Review by mitchmindtree — APPROVED** [review]
Nice, I think this should make for a good start!

Currently this only tests the default init type (`--contract`) but we can extend this with each of the different init types (i.e. `--contract`, `--library`, `--predicate`, `--script`) in a follow-up PR. This could be a good use-case for the CI matrix feature, e.g. https://github.com/FuelLabs/fuel-core/commit/344115aa6c6a18863496cdf2dc6bb6e3fba05e94#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR57

**Comment by ra0x3** [comment]
> Nice, I think this should make for a good start!

@mitchmindtree 
- I agree we should add the other implementation as well eventually. I can create this as a `goodfirstissue` 
- I see that Github actions allows for parallel jobs via the `matrix` - we should definitely parallelize what we can to speed up CI (not super hugely important however)
- My question here is -- I see many of our CI jobs that seem to be independent of other jobs (this job I just merged for example). Is there any reason we're not using a `matrix` strategy on these jobs currently?

**Pull request merged**